patches/minecraft/net/minecraft/world/entity/monster/EnderMan.java.patch

```diff
@@ -16,12 +16,15 @@
     }
  
     protected void m_8097_() {
-@@ -194,7 +_,7 @@
+@@ -194,7 +_,10 @@
  
     boolean m_32534_(Player p_32535_) {
        ItemStack itemstack = p_32535_.m_150109_().f_35975_.get(3);
 -      if (itemstack.m_150930_(Blocks.f_50143_.m_5456_())) {
-+      if (net.minecraftforge.common.ForgeHooks.shouldSuppressEnderManAnger(this, p_32535_, itemstack)) {
++      net.minecraftforge.eventbus.api.Event.Result result = net.minecraftforge.common.ForgeHooks.onEnderManAnger(this, p_32535_);
++      if (result != net.minecraftforge.eventbus.api.Event.Result.DEFAULT) {
++         return result == net.minecraftforge.eventbus.api.Event.Result.ALLOW;
++      } else if (itemstack.isEnderMask(p_32535_, this)) {
```

Comment:
Why did you pull out this logic into the patch instead of just changing `ForgeHooks.shouldSuppressEnderManAnger`?

Reply:
because just a boolean can not be used to determine what to do because ALLOW/DENY means it skips the vanilla distance and line of sight checks and forges isEnderMask check

Reply:
`EnderMan#isLookingAtMe` returns a boolean though:
```
if (ForgeHooks.shouldCancelEndermanAnger(this, stack, player)) {
    return false;
} else { 
 ....
 }
```